File: src/main/java/com/project/networktechnologiesproject/controller/dto/RegisterDto.java
package com.project.networktechnologiesproject.controller.dto;

import com.project.networktechnologiesproject.commonTypes.UserRole;
import org.apache.catalina.User;

public class RegisterDto {
    private String password;
    private String username;
    private UserRole role;
    private String email;

    public RegisterDto() {
    }

    public RegisterDto(String password, String username, UserRole role, String email) {
        this.password = password;
        this.username = username;
        this.role = role;
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public UserRole getRole() {
        return role;
    }

    public void setRole(UserRole role) {
        this.role = role;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: src/api/dto/register/register.dto.ts
export class RegisterDto {
  password: string = '';
  username: string = '';
  role: Role = 'ROLE_READER';
  email: string = '';
}

export type Role = 'ROLE_ADMIN' | 'ROLE_READER';

export class RegisterResponseDto {
  username: string = '';
  role: Role = 'ROLE_READER';
  email: string = '';
}
